### ISSUE TITLE
redhat 7.3 diskful install xcatprobe osdeploy reports failure.

### ISSUE THREAD

**Issue by ralphbellofatto** [xcat2/xcat-core#2176]
Using the example setup for rh7.3 supplied by xcat when we perform the copycds we get the following failure:

```
==================osdeploy_probe_report=================
There is 1 node provision failures
c460login02 : Without provision process during rollback time window                           [FAIL]

# howerver lsdef status shows:
lsdef c460login02 -i status -c
c460login02: status=booted

```

We did the following to set this up.

```
copycds  \
   /root/rh73/RHEL-7.3-20161019.0-Server-ppc64le-dvd1.iso \
    -p /install/RHEL-7.3-20161019.0-Server/ppc64le
```

```
nodeset': nodeset c460login02 osimage=rhels7.3-ppc64le-install-compute
xdsh c460login02 reboot
xcatprobe osdeploy -n c460login02osdeploy
xcatprobe osdeploy -n c460login02 | tee /tmp/c460login02.xcatprobe.osdeploy.log

```

and this shows:
```
==================osdeploy_probe_report=================
There is 1 node provision failures
c460login02 : Without provision process during rollback time window                           [FAIL]
```

attached are the log file, the output of the xcatprobe and the console log for this event:
[rh73-prov-err.zip](https://github.com/xcat2/xcat-core/files/601569/rh73-prov-err.zip)

**Comment by xuweibj** [comment]
Hi @ralphbellofatto , 

In our code, considers CN's restart as the start of os provision. If there is no log of restart, will show the error message ``Without provision process during rollback time window``.
And now only ``rpower``, ``rnetboot`` and ``rinstall`` are considered as restart.
But here use ``xdsh c460login02 reboot`` to restart CN, so did not show correct message.

We will discuss how to solve this issue.

**Comment by ralphbellofatto** [comment]
What is the command for a compute node "restart"??

**Comment by xuweibj** [comment]
@ralphbellofatto , sorry maybe I did not explain well.

In our environment, after ``nodeset`` command, we usually use ``rpower  c460login02 boot`` or ``rpower c460login02 reset`` on MN to restart CN. If the arch of CN is ppc64, use ``rnetboot``. 

In this case, you used ``xdsh c460login02 reboot``, so I just want to know whether there is other command you usually use.

**Comment by hu-weihua** [comment]
@ralphbellofatto ,  when we designed the ``osdeploy``, we assumed customer hope to diagnose what happened when they deploy brand new server.  So in this case the only way they can use to boot server is using ``xCAT`` remote control command ``rpower`` (except boot server manually or they can do remote control by leveraging other tools except ``xCAT``, these two scenarios we can't support. at least it's true now).   But in your case, you redeploy a server which has been install xCAT before, so you can reboot this server by ``xdsh``.  

How we judge a provision process start now is when we capture below commands

     if ($command eq "rinstall" or $command eq "rnetboot" or ($command eq "rpower" and $sub_command =~ /on|boot|reset/)) {

So in your case, we missed the operation ``xdsh <node> reboot``, so we reported ``Without provision process during rollback time window ``. 

But we think it make sense to redeploy a server which has been installed xCAT before.  We plan to add ``xdsh <node> reboot`` to our code.  Do you have any comment about this or you think there still are other use scenarios? your rich practical experience are very helpful for us to make ``osdeploy`` more useful. thanks.

**Comment by ralphbellofatto** [comment]
Is it possible for that very last message to reflect the status=failed,status=booted field that is in the lsdef -i status?

**Comment by ralphbellofatto** [comment]
also, you you may not need to look at the command itself xdsh xxx reboot, but instead look for the message that comes back when the shutdown happens.

You are already displaying that status in the lsdef -i status field as the node reboots, so that actually sounds like the best place to detect that transition.

**Comment by hu-weihua** [comment]
@ralphbellofatto , sorry for replay later.  

We have ever taken these status into account, such like  ``powering-on`` and ``powering-off``. But unfortunately,  we don't always have these status in the log when we trigger reboot through different command. The other thing is we allow customer to configure if show these status in site table.  So after investigate all the platform and scenario, we choose capturing reboot command. 

I didn't get the point of ``Is it possible for that very last message to reflect the status=failed,status=booted field that is in the lsdef -i status?``.   if ``status=booted``, we have other log ``provision completed``, all of these log identify the provision successful.   Do you think customer are more familiar with ``booted`` to  identify the provision successful?

**Comment by ralphbellofatto** [comment]
I thought the purpose of the command "xcatprobe osdeploy" was to report the progress of the boot.  The boot here was successful, yet the osprobe deploy reported it as failure.

the user has to then know that the failure reported by osdeploy may be wrong and subsequently check the lsdef -i status for that node.  That seems broken to me...

**Comment by whowutwut** [comment]
@ralphbellofatto  If you have a chance to verify this, please feel free to close, otherwise @tingtli  Can you help find someone in FVT to validate this fix so we can close the issue?

**Comment by whowutwut** [comment]
@xuweibj @hu-weihua  I'm looked more closely at this issue and it seems like there was no code changes resulting from this.  Was the pending added by mistake? I'm going to move this to future release...  we either need to agree we will not fix this or agree on providing something that makes sense based on the feedback from @ralphbellofatto

**Comment by ralphbellofatto** [comment]
If we provide this feature to probe the os deployment it should not be reporting a failure that then the user has to know to ignore.

**Comment by xuweibj** [comment]
@whowutwut I'm sorry, I forgot fill this issue number in #2308.
@ralphbellofatto , I think for this issue, it should not report failure. Do you mean this should report failure? I feel a little confused.

**Comment by ralphbellofatto** [comment]
if the boot was successful the osprobe should report success, as of when this was written, a successful boot at our site was reported as failure.

There is another bug report, where we have the final xcat status report for the post scripts reports failed, and the osprobe fails to show the error.

**Comment by xuweibj** [comment]
Hi @ralphbellofatto , which version do you use now? We have tested many times, do you still use ``xdsh node reboot`` command?

**Comment by ralphbellofatto** [comment]
xdsh node reboot, yes...

I have not tested it since it filed this bug report...

**Comment by xuweibj** [comment]
@whowutwut We have fixed this and #2177 in 2.13.1, and FVT have verified. @ralphbellofatto Do you want to verify it? Or our FVT will close this issue.

**Comment by whowutwut** [comment]
@xuweibj If you have validated it, can you paste the new behavior in here to demonstrate the external behavior changes that have been made. Then Ralph can validate if it's sufficient to close the issue.
